tinyagentos/shortcuts/token_source.py

```
@@ -0,0 +1,107 @@
+    for k in to_remove:
+        del _cache[k]
+
+
+def _exec_token_source(
+    agent_name: str, source: dict[str, Any], kind: str
+) -> Optional[str]:
+    container = f"taos-agent-{agent_name}"
+
+    if kind == "container_env":
+        var = source["var"]
+        result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)